pontoon/base/views.py

```diff
@@ -118,21 +117,27 @@ def locale_admin(request, locale):
                 return HttpResponseBadRequest(e)
 
         if current != new:
-            l.translators_group.user_set = User.objects.filter(pk__in=new)
-            l.translators_group.save()
+            group.user_set = User.objects.filter(pk__in=new)
+            group.save()
+
+    if request.method == 'POST':
+        update_group(l.translators_group, 'translators')
+        update_group(l.managers_group, 'managers')
 
-    users_translators = l.translators_group.user_set.all()
-    users_all = User.objects.exclude(pk__in=users_translators).exclude(email="")
-    users_contributors = User.translators.filter(translation__locale=l).distinct()
+    managers = l.managers_group.user_set.all()
+    translators = l.translators_group.user_set.exclude(pk__in=managers).all()
```

Comment:
Is there a way you could filter by the group itself instead of the managers subquery? Like `group_set=l.managers_group` or something. I'd expect that to be faster than the subquery, which may end up mattering at high user counts.